pkg/platform/api/graphql/model/buildplanner/buildExpression.go

```diff
@@ -0,0 +1,267 @@
+package model
+
+import (
+	"encoding/json"
+	"errors"
+	"time"
+
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/logging"
+)
+
+type Comparator string
+
+type Operation int
+
+const (
+	ComparatorEQ  Comparator = "eq"
+	ComparatorGT             = "gt"
+	ComparatorGTE            = "gte"
+	ComparatorLT             = "lt"
+	ComparatorLTE            = "lte"
+	ComparatorNE             = "ne"
+
+	OperationAdd Operation = iota
+	OperationRemove
+	OperationUpdate
+
+	SolveFuncName           = "solve"
+	SolveLegacyFuncName     = "solve_legacy"
+	RequirementsKey         = "requirements"
+	AtTimeKey               = "at_time"
+	RequirementNameKey      = "name"
+	RequirementNamespaceKey = "namespace"
+)
+
+func (o Operation) String() string {
+	switch o {
+	case OperationAdd:
+		return "add"
+	case OperationRemove:
+		return "remove"
+	case OperationUpdate:
+		return "update"
+	default:
+		return "unknown"
+	}
+}
+
+var funcNodeNotFoundError = errors.New("Could not find function node")
+
+type Requirement struct {
+	Name               string               `json:"name"`
+	Namespace          string               `json:"namespace"`
+	VersionRequirement []VersionRequirement `json:"version_requirements,omitempty"`
+}
+
+type VersionRequirement map[Comparator]string
+
+type BuildExpression map[string]interface{}
+
+func NewBuildExpression(data []byte) (BuildExpression, error) {
+	var bx BuildExpression
+
+	err := json.Unmarshal(data, &bx)
+	if err != nil {
+		return nil, errs.Wrap(err, "Could not unmarshal JSON")
+	}
+
+	return bx, nil
+}
+
+func (bx BuildExpression) Requirements() ([]Requirement, error) {
+	requirementsNode, err := getRequirementsNode(bx)
+	if err != nil {
+		return nil, errs.Wrap(err, "Could not get requirements node")
+	}
+
+	requirementsData, err := json.Marshal(requirementsNode)
```

Comment:
I do not know of a way to do this directly. The json marshaling/unmarshaling is a cheap way to do this. I tried to use a package like `mapstructre` but that didn't capture all of the data.

Reply:
That's okay. I suspect there's a way to do it, but no use wasting time doing so. Just add a comment about this so we can potentially revisit in the future (perhaps if we try to consolidate our approaches).

Reply:
@daved can you weigh in on this?

Reply:
As we talked about in stand, this would require a conversion function. In a way, this is a light abuse of the marshaling/unmarshaling behavior, but it seems acceptable to everyone at the moment.